### PR TITLE
DR-3297: Rewrite getHomePageData to no longer use `items/counts`

### PR DESCRIPTION
**Pull request by 7emansell**
## Ticket:

- JIRA ticket [DR-3297](https://newyorkpubliclibrary.atlassian.net/browse/DR-3297)

## This PR does the following:

- Adds new `getItemsCountFromCollections()` which hits `/collections/{uuid}`  to get `numItems` for each passed `uuid` (really only a helper for `getHomePageData`)
- NOTE: this is declared and tested in a separate file because of Jest mocking conventions. When I get to this https://newyorkpubliclibrary.atlassian.net/browse/DR-3271 I'll refactor all of the API helper tests and it'll be less weird

## How has this been tested? How should a reviewer test this?

Go to the homepage and think to yourself, is this slower than usual 

### Checklist:

- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.
- [x] I have updated the CHANGELOG.md.

[DR-3297]: https://newyorkpubliclibrary.atlassian.net/browse/DR-3297?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by avertrees — APPROVED** [review]

File: app/src/components/pages/divisionsPage/divisionsPage.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ export default function DivisionsPage({ summary, divisions }: DivisionsProps) {
       ]}
       adobeAnalyticsPageName={createAdobeAnalyticsPageName("divisions")}
     >
-      {divisions && divisions.length > 0 ? (
+      {divisions && divisions?.length > 0 ? (
```
Comment (EdwinGuzman) [inline app/src/components/pages/divisionsPage/divisionsPage.tsx:36]: wow sneaky?

File: app/src/utils/itemCount.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Returns a map of collection uuids with corresponding item counts.
+ * @param {string[]} uuids - collection uuids
+ */
+export const getItemsCountFromCollections = async (uuids: string[]) => {
+  const results = {};
+  const apiUrl = `${process.env.API_URL}/api/v2/collections/`;
+  await Promise.all(
+    uuids.map(async (uuid) => {
```
Comment (EdwinGuzman) [inline app/src/utils/itemCount.ts:11]: This is me being nitpicky but the syntax is interesting. This works so no need to change but I would generally prefer to create the list of unresolved promises in a variable and then pass it to the `Promise.all` call. Just minor readability comment.

**Review by EdwinGuzman — APPROVED** [review]
Is the backend updating their cached time concurrently with this update? It'd be good to just test this to keep all the different variable separate so we know what works.

**Comment by 7emansell** [comment]
Backend is making no changes until we see the outcome of this update as far as I know @EdwinGuzman

File: app/src/utils/itemCount.ts (hunk 2 above)
Comment (7emansell) [reply app/src/utils/itemCount.ts]: Is this what you're thinking ?

File: app/src/components/pages/divisionsPage/divisionsPage.tsx (hunk 1 above)
Comment (avertrees) [reply app/src/components/pages/divisionsPage/divisionsPage.tsx:36]: yes

Comment (avertrees) [reply app/src/components/pages/divisionsPage/divisionsPage.tsx:36]: as per my request @EdwinGuzman

**Review by EdwinGuzman — APPROVED** [review]
Yep, looks good

**Pull request merged**